hack/lib/golang.sh

```diff
@@ -972,3 +972,14 @@ kube::golang::build_binaries() {
     done
   fi
 }
+
+# Returns a go package pattern to match all packages in the workspace
+#
+# Use like:
+# kube::util::read-array targets < <(kube::golang::workspace_all)
+#
+# Doing it this way is MUCH faster than simply saying "all", and there doesn't
+# seem to be a simpler way to express "this whole workspace".
+kube::golang::workspace_all() {
+  go list -f '{{.Dir}}/...' -m
```

Comment:
This gives absolute paths, but is otherwise equivalent to `go work edit -json | jq -r '.Use[].DiskPath + "/..."'` without requiring jq, making it marginally simpler / faster in scripts that otherwise wouldn't require jq.

Haven't yet found a really great way to produce _exactly_ the same output with go list but this seems OK

Reply:
In module mode we're formatting https://pkg.go.dev/cmd/go/internal/modinfo#ModulePublic

Reply:
Using `.Dir` here produces absolute paths, which works but seems like a recipe for something to get messed up.  Using `.Path` produces module+pkg paths, which seems safer?

That said, I see some errors:

```
$ go list `go work edit -json | jq -r '.Use[].DiskPath + "/..."'` | wc -l
2492

$ go work edit -json | jq -r '.Use[].DiskPath + "/..."' | wc -l
31

$ go list $(go work edit -json | jq -r '.Use[].DiskPath + "/..."') | wc -l
2492

$ go list -m -f '{{.Path}}/...' | wc -l
31

$ go list $(go list -m -f '{{.Path}}/...') | wc -l
warning: ignoring symlink /home/thockin/src/kubernetes/cluster/gce/cos
warning: ignoring symlink /home/thockin/src/kubernetes/cluster/gce/custom
warning: ignoring symlink /home/thockin/src/kubernetes/cluster/gce/ubuntu
warning: ignoring symlink /home/thockin/src/kubernetes/third_party/etcd
warning: ignoring symlink /home/thockin/src/kubernetes/cluster/gce/cos
warning: ignoring symlink /home/thockin/src/kubernetes/cluster/gce/custom
warning: ignoring symlink /home/thockin/src/kubernetes/cluster/gce/ubuntu
warning: ignoring symlink /home/thockin/src/kubernetes/third_party/etcd
2492

$ go list ./... | wc -l
1247

$ go list k8s.io/kubernetes/... | wc -l
warning: ignoring symlink /home/thockin/src/kubernetes/cluster/gce/cos
warning: ignoring symlink /home/thockin/src/kubernetes/cluster/gce/custom
warning: ignoring symlink /home/thockin/src/kubernetes/cluster/gce/ubuntu
warning: ignoring symlink /home/thockin/src/kubernetes/third_party/etcd
warning: ignoring symlink /home/thockin/src/kubernetes/cluster/gce/cos
warning: ignoring symlink /home/thockin/src/kubernetes/cluster/gce/custom
warning: ignoring symlink /home/thockin/src/kubernetes/cluster/gce/ubuntu
warning: ignoring symlink /home/thockin/src/kubernetes/third_party/etcd
1247
```

I am fine either way, as long as we handle the errors cleanly.  We depend on JQ in a lot of places already.

Reply:
> Using .Path produces module+pkg paths, which seems safer?

I did this first but we have a few places that assume we're going to use directory paths instead of import paths.

> I am fine either way, as long as we handle the errors cleanly. We depend on JQ in a lot of places already.

Yeah, I just wanted to avoid the inner require, but if we really care about that we could cache the check-for-jq result ...